src/test/java/uk/gov/digital/ho/pttg/application/namematching/candidates/SpecialCharactersTest.java

```diff
@@ -0,0 +1,181 @@
+package uk.gov.digital.ho.pttg.application.namematching.candidates;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.digital.ho.pttg.application.namematching.CandidateName;
+import uk.gov.digital.ho.pttg.application.namematching.InputNames;
+
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SpecialCharactersTest {
+
```

Comment:
Also what do we do about special characters in aliases?

Reply:
Currently nothing but it feels like we should handle in a similar manner to non-alias names. I have asked @ashanperera89 to confirm this requirement.

Reply:
Confirmed by @ashanperera89 that we should remove when the name has aliases, we should include them in the SpecialCharacter treatment. As is the case when we don't have special characters, the ordering of candidates should be different depending on whether or not aliases exist, and should follow the ordering rules already implemented.